src/checkout-buttons/strategies/braintree/braintree-paypal-button-strategy.ts

```diff
@@ -45,15 +46,17 @@ export default class BraintreePaypalButtonStrategy implements CheckoutButtonStra
 
         this._braintreeSDKCreator.initialize(paymentMethod.clientToken);
         const container = `#${options.containerId}`;
+        const messagingContainer = options.braintreepaypal?.messagingContainer;
 
         this._renderButtonsData = {
             paymentMethod,
             paypalOptions,
             container,
+            messagingContainer,
         };
 
         return Promise.all([
-            this._braintreeSDKCreator.getPaypalCheckout({currency: currency?.code }, (paypalCheckoutInstance: PaypalClientInstance) => this.renderButtons(paypalCheckoutInstance)),
+            this._braintreeSDKCreator.getPaypalCheckout({currency: currency?.code, components: 'buttons,messages' }, (paypalCheckoutInstance: PaypalClientInstance) => this.renderButtons(paypalCheckoutInstance)),
```

Comment:
i don't like idea with passing string for components, could we separate them as array or may be introduce own constant for them?
const paypalComponents = ['buttons', 'messages']; and you could directly declare it in, without needs in passing here
src/payment/strategies/braintree/braintree-sdk-creator.ts

Reply:
Fixed